docs/specification/sep/sep-client-id-metadata.md

```diff
@@ -0,0 +1,270 @@
+3. **This prevents redirect URI manipulation attacks** common with self-asserted registration
+
+
+### Risks of this approach
+
+#### Risk: Localhost URL Impersonation
+
+A limitation of Client ID Metadata Documents is that they cannot prevent localhost URL impersonation by itself. An attacker can claim to be any client by:
+1. Providing the legitimate client's metadata URL as their client_id
+2. Binding to the same localhost port the legitimate client uses
+3. Intercepting the authorization code when the user approves
+
+This attack is concerning because the server sees the correct metadata
+document and the user sees the correct client name, making detection
+difficult. While platform-specific attestations (iOS DeviceCheck, Android 
+Play Integrity) could address this, they're not universally available. An 
+approach using JWKS and short-lived JWTs signed by a server-side component 
+of the client could raise the cost of attack but cannot eliminate it 
+entirely - an attacker could still proxy requests to the legitimate 
+client's signing endpoint. 
+
+Fully mitigating this risk is outside the scope of this proposal. This
+proposal has the same risks as DCR does in a localhost redirect scenario.
+
+Servers SHOULD display additional warnings for localhost-only clients.
+
+#### Risk: Server Side Request Forgery (SSRF)
+
+The authorization server takes a URL as input from an unknown client, and then fetches that URL. A malicious client could use this to send non-metadata requests on behalf of the authorization server. An example would be sending a URL corresponding to a private administration endpoint that the authorization server has access to.
+
+This can be prevented by validating the URL's and the IP's those URL's resolve to prior to initiating a fetch request.
+
+
+#### Risk: Distributed Denail of Service (DDoS)
+
+Similarly, an attacker could try to leverage a pool of authorization servers to perform a denial of service attack on a non-MCP server.
+
+There is not any additional amplification for the fetch request (i.e. the bandwidth from the client to make the request roughly equals the bandwidth of the request sent to the target server), and each authorization server can aggressively cache the result of these metadata fetches, so it is unlikely to be an attractive DDoS vector.
+
+
+#### Risk: Maturity of referenced specification
+
+The RFC for Client ID Metadata documents is still a draft. It has been implemented by the platform Bluesky, but has not been ratified or very widely adopted outside of that.
+
+This has the risk that there are implementation challenges or flaws in the protocol that have not surfaced yet. However, even though DCR has been ratified, and it also has a number of implementation challenges that developers are facing when trying to use it in an open ecosystem context like MCP.  Those challenges are the motiviation behind this proposal.
```

Comment:
Probably worth adding something similar to BlueSky's call-out about the draft:

> The Client ID Metadata Document specification ([draft-parecki-oauth-client-id-metadata-document](https://datatracker.ietf.org/doc/draft-parecki-oauth-client-id-metadata-document/) is still a draft and may evolve over time. Our intention is to evolve and align with subsequent drafts and any final standard, while minimizing disruption and breakage with existing implementations.

https://atproto.com/specs/oauth